### DISCUSSION TITLE
Help with overriding/setting comment syntax in cpp files...

### DISCUSSION THREAD

**Discussion by mellery451** [nvim-treesitter/nvim-treesitter#6569] [Q&A]
I'm aware that we seem to have support for doxygen and re2c in cpp comments. I'd like to add another local option for rst syntax override if my comment contains the text "SPECIFICATION BEGIN" (I can refine sentinel text once I have something working)...so I created a file at .config/nvim/after/queries/cpp/injections.scm with contents:

     ((comment) @injection.content
         (#lua-match? @injection.content "SPECIFICATION BEGIN")
         (#set! injection.language "rst"))

..which is not working. Are there other files I need to add/update? Also, what determines the order that these matchers are evaluated? Could the doxygen matcher be overriding my custom one?

**Comment by clason** [comment]
`:h treesitter-query-modeline-extends`

and the order is "all are applied in order, and last one sticks"

**Comment by mellery451** [comment]
I added that - still no luck, but I think my TS is dealing with comments in cpp files very oddly. No matter what I do, InspectTree does not show my comment block as (comment)...I usually get something like (brief_header) (brief_description), but sometimes even that is not there and the comment block is totally ignored by TS. strange.

**Comment by mellery451** [comment]
I'm still banging my head against the wall. Here is what I have now in my cpp injection file:

```
;; extends
;; inherits: cpp

((comment) @injection.content
  (#lua-match? @injection.content "^//! ")
  (#offset! @injection.content 0 4 0 0)
  (#set! injection.language "rst"))
```

and what I get is: 

![Screenshot 2024-05-22 at 4 18 17 PM](https://github.com/nvim-treesitter/nvim-treesitter/assets/463430/d16211f3-24cd-4e5d-859f-aeb9011d7678)

Why doesn't ts merge all adjacent comment lines into a single comment entity? seems like that would help to denote complete sublanguage documents, but maybe I misunderstand.

What else should I try?

**Reply by clason** [reply]
`injection.combined`

**Reply by mellery451** [reply]
![Screenshot 2024-05-23 at 9 02 29 AM](https://github.com/nvim-treesitter/nvim-treesitter/assets/463430/69d5378a-156f-49a0-abe4-d0d7e9d76195)

```
;; extends
;; inherits: cpp

((comment) @injection.content
  (#lua-match? @injection.content "^//! ")
  (#offset! @injection.content 0 4 0 0)
  (#set! injection.combined)
  (#set! injection.language "rst"))
```

no change it seems like

**Reply by clason** [reply]
Well, yes, since in this case every comment line is its own node. Try `(comment)+`.

**Reply by mellery451** [reply]
(comment)+ didn't change anything for me, except that I needed to remove the offset spec otherwise I get errors. Nothing I do makes these comment lines unify. Maybe that doesn't matter in the end...but RST just won't apply no matter what I try.
